src/main/java/seedu/address/model/person/Birthday.java

```diff
@@ -0,0 +1,257 @@
+        requireNonNull(birthday);
+
+        if (birthday.isEmpty()) {
+            return true;
+        }
+
+        int[] processedSplitDate;
+
+        try {
+            processedSplitDate = processDate(birthday);
+        } catch (NumberFormatException nfe) {
+            return false;
+        }
+
+        if (processedSplitDate.length != 3 || !isValidDate(processedSplitDate)) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the date is valid. A date is valid if it is possible for it to exist.
+     */
+    private static boolean isValidDate(int[] processedSplitDate) {
+        requireNonNull(processedSplitDate);
+        if (isLeapYearDate(processedSplitDate)) {
+            return true;
+        }
+        if (isYearValid(processedSplitDate[DATE_YEAR_INDEX])
+                && isDayAndMonthValid(processedSplitDate[DATE_DAY_INDEX], processedSplitDate[DATE_MONTH_INDEX])) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if the month and day is valid.
+     * The month and day is valid if the day can exist in the specific month.
+     */
+    private static boolean isDayAndMonthValid(int day, int month) {
+        if (!isMonthValid(month)) {
+            return false;
+        }
+
+        final int dayUpperLimitForMonth = MONTH_TO_DAY_MAPPING[month - ZERO_BASED_ADJUSTMENT];
+
+        if (!isDayValid(dayUpperLimitForMonth, day)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if the day is valid.
+     * A day is valid if it is at least 1 and smaller than the largest possible day for the specific month.
+     */
+    private static boolean isDayValid(int dayUpperLimitForMonth, int day) {
+        if (day < SMALLEST_POSSIBLE_DAY || day > dayUpperLimitForMonth) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the month is valid.
+     */
+    private static boolean isMonthValid(int month) {
+        if (month < SMALLEST_POSSIBLE_MONTH || month > LARGEST_POSSIBLE_MONTH) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the year is valid.
+     * A year is valid if has at least 4 digits.
+     */
+    private static boolean isYearValid(int year) {
+        if (year < SMALLEST_ALLOWED_YEAR || year > LARGEST_ALLOWED_YEAR) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true if the date is a leap year day.
+     * A leap year day must be on 29 April.
+     */
+    private static boolean isLeapYearDate(int[] processedSplitDate) {
+        if (processedSplitDate[DATE_DAY_INDEX] == LEAP_YEAR_DAY) {
+            if (processedSplitDate[DATE_MONTH_INDEX] == LEAP_YEAR_MONTH_FEBRUARY) {
+                if (isLeapYear(processedSplitDate[DATE_YEAR_INDEX])) {
+                    return true;
+                }
```

Comment:
Avoid this "arrow" pattern with nested-if. You can just combine the boolean expressions:
    if (processedSplitDate[DATE_DAY_INDEX] == LEAP_YEAR_DAY
        && processedSplitDate[DATE_MONTH_INDEX] == LEAP_YEAR_MONTH_FEBRUARY
        && isLeapYear(processedSplitDate[DATE_YEAR_INDEX]))
        return true;
    }

Reply:
But wouldnt combining be longer? Like it becomes hard to read

Reply:
It is still doing the same thing, so nested-if isn't necessary. It still has the same number of lines and I can still read the expressions easily.